python/ml4ir/applications/ranking/tests/test_ml4ir_sanity.py

```diff
@@ -0,0 +1,266 @@
+        model_config_file = pathlib.Path(__file__).parent / "data" / "configs" / "model_config_sanity_tests.yaml"
+        train_ml4ir(p.as_posix(), feature_config_file.as_posix(), model_config_file.as_posix(), log_dir.as_posix())
+        logisticReg = LogisticRegression(fit_intercept=False, random_state=args.seed, max_iter=100,
+                                warm_start=True)
+        perceptron = Perceptron(fit_intercept=False, max_iter=100, random_state=args.seed, validation_fraction=0.0001,
+                       warm_start=True)
+        ml4ir_weights = pd.read_csv("models/test_command_line/coefficients.csv")["weight"].tolist()
+        ml4ir_weights.append(0.0)
+        ml4ir = wrap_linear_model_in_sklearn_model(df, ml4ir_weights)
+        train_sklearn_model(df, logisticReg, featureCount=args.n_features)
+        train_sklearn_model(df, perceptron, featureCount=args.n_features)
+
+        ml4ir_mrr = check_MRR(df, ml4ir, args.n_features)
+        perceptron_mrr = check_MRR(df, perceptron, args.n_features)
+        log_regression_mrr = check_MRR(df, logisticReg, args.n_features)
+        return ml4ir_mrr, perceptron_mrr, log_regression_mrr
+
+
+def train_ml4ir(data_dir, feature_config, model_config, logs_dir):
+
+    argv = ['--data_dir', data_dir, '--feature_config', feature_config, '--loss_type', "pointwise", '--scoring_type', "pointwise",
+            '--run_id', 'test_command_line', '--data_format', 'csv', '--execution_mode', 'train_inference_evaluate', '--loss_key', 'sigmoid_cross_entropy',
+            '--num_epochs', '150', '--model_config', model_config, '--batch_size', '16', '--logs_dir', logs_dir,
+            '--max_sequence_size', '25', '--train_pcent_split', '1.0', '--val_pcent_split', '-1', '--test_pcent_split', '-1', '--early_stopping_patience', '1000']
+    args = get_args(argv)
+    rp = RankingPipeline(args=args)
+    rp.run()
+
+
+class TestML4IRSanity(unittest.TestCase):
+    def setUp(self):
+        self.dir = pathlib.Path(__file__).parent
+        self.p = pathlib.Path(self.dir) / "data" /"ml4ir_sanity_test_working_dir"
+        self.p.mkdir(parents=True, exist_ok=True)
+        self.log_dir = self.p / 'logs'
+        self.log_dir.mkdir(parents=True, exist_ok=True)
+        train = self.p / 'train'
+        train.mkdir(parents=True, exist_ok=True)
+        test = self.p / 'test'
+        test.mkdir(parents=True, exist_ok=True)
+        validation = self.p / 'validation'
+        validation.mkdir(parents=True, exist_ok=True)
+
+    def tearDown(self):
+        shutil.rmtree(self.p)
+
+    def test_ml4ir_sanity_1(self):
+        args = Namespace(n_classes=2, n_samples=1000, n_features=2, n_redundant=0, n_informative=2,
+                         n_clusters_per_class=2, flip_y=-1, class_sep=1.0, seed=13, n_trials=100)
+        ml4ir_mrr, perceptron_mrr, log_regression_mrr= test_ml4ir_sanity_pipeline(args, self.p, self.log_dir)
+        assert ml4ir_mrr >= perceptron_mrr
+        assert ml4ir_mrr >= log_regression_mrr
```

Comment:
I don't get why we expect it to be better actually. Shouldn't we check if it is `np.isclose(atol=0.1)`?

Reply:
Same for all tests.